crates/ide-assists/src/handlers/into_to_qualified_from.rs

```diff
@@ -52,17 +52,25 @@ pub(crate) fn into_to_qualified_from(acc: &mut Assists, ctx: &AssistContext<'_>)
         == FamousDefs(sema, scope.krate()).core_convert_Into()?
     {
         let type_call = sema.type_of_expr(&method_call.clone().into())?;
-        let type_call_disp =
-            type_call.adjusted().display_source_code(db, scope.module().into(), true).ok()?;
+        let adjusted_tc = type_call.adjusted();
 
+        if adjusted_tc.contains_unknown() {
+            return None;
+        }
+
+        let sc = adjusted_tc.display_source_code(db, scope.module().into(), true).ok()?;
         acc.add(
             AssistId("into_to_qualified_from", AssistKind::Generate),
             "Convert `into` to fully qualified `from`",
             nameref.syntax().text_range(),
             |edit| {
                 edit.replace(
                     method_call.syntax().text_range(),
-                    format!("{}::from({})", type_call_disp, receiver),
+                    if sc.chars().find(|c| !c.is_alphanumeric() && c != &':').is_some() {
```

Comment:
```suggestion
                    if sc.chars().any(|&c| !c.is_alphanumeric() && c != ':') {
```
Alternatively (with the then and else branches flipped)
```rs
                    if sc.chars().all(|&c| c.is_alphanumeric() || c = ':') {
                        format!("{}::from({})", sc, receiver)
                    } else {
                        format!("<{}>::from({})", sc, receiver)
                    },
```
would be a bit more readable due to less negations